adapters/repos/db/lsmkv/memtable_flush.go

```diff
@@ -46,46 +45,47 @@ func (m *Memtable) flush() error {
 	if err != nil {
 		return err
 	}
-
-	w := bufio.NewWriter(f)
+	segmentFile := segmentindex.NewSegmentFile(
+		segmentindex.WithBufferedWriter(bufio.NewWriter(f)),
+	)
 
 	var keys []segmentindex.Key
 	switch m.strategy {
 	case StrategyReplace:
-		if keys, err = m.flushDataReplace(w); err != nil {
+		if keys, err = m.flushDataReplace(segmentFile); err != nil {
 			return err
 		}
 
 	case StrategySetCollection:
-		if keys, err = m.flushDataSet(w); err != nil {
+		if keys, err = m.flushDataSet(segmentFile); err != nil {
 			return err
 		}
 
 	case StrategyRoaringSet:
-		if keys, err = m.flushDataRoaringSet(w); err != nil {
+		if keys, err = m.flushDataRoaringSet(segmentFile); err != nil {
 			return err
 		}
 
 	case StrategyMapCollection:
-		if keys, err = m.flushDataMap(w); err != nil {
+		if keys, err = m.flushDataMap(segmentFile); err != nil {
 			return err
 		}
 
 	default:
 		return fmt.Errorf("cannot flush strategy %s", m.strategy)
 	}
 
-	indices := &segmentindex.Indexes{
+	indexes := &segmentindex.Indexes{
 		Keys:                keys,
 		SecondaryIndexCount: m.secondaryIndices,
 		ScratchSpacePath:    m.path + ".scratch.d",
 	}
 
-	if _, err := indices.WriteTo(w); err != nil {
+	if _, err := segmentFile.WriteIndexes(indexes); err != nil {
 		return err
 	}
 
-	if err := w.Flush(); err != nil {
```

Comment:
it gives me the impression that `flush` is not called into the underlying buffered writter `w`, is it called somewhere else?

Reply:
Yes, now it is always called at the end of [WriteChecksum](https://github.com/weaviate/weaviate/blob/feature/segment-checksums/adapters/repos/db/lsmkv/segmentindex/segment_file.go#L173-L175)